GPU/Common/SplineCommon.cpp

```diff
@@ -908,13 +908,22 @@ void DrawEngineCommon::SubmitSpline(const void *control_points, const void *indi
 		const bool hasColor = (origVertType & GE_VTYPE_COL_MASK) != 0;
 		const bool hasTexCoords = (origVertType & GE_VTYPE_TC_MASK) != 0;
 
-		tessDataTransfer->PrepareBuffers(pos, tex, col, count_u * count_v, hasColor, hasTexCoords);
+		int posStride, texStride, colStride;
+		tessDataTransfer->PrepareBuffers(pos, tex, col, posStride, texStride, colStride, count_u * count_v, hasColor, hasTexCoords);
+		float *p = pos;
+		float *t = tex;
+		float *c = col;
 		for (int idx = 0; idx < count_u * count_v; idx++) {
-			memcpy(pos + idx * 4, points[idx]->pos.AsArray(), 3 * sizeof(float));
-			if (hasTexCoords)
-				memcpy(tex + idx * 4, points[idx]->uv, 2 * sizeof(float));
-			if (hasColor)
-				memcpy(col + idx * 4, Vec4f::FromRGBA(points[idx]->color_32).AsArray(), 4 * sizeof(float));
+			memcpy(p, points[idx]->pos.AsArray(), 3 * sizeof(float));
+			p += posStride;
+			if (hasTexCoords) {
+				memcpy(t, points[idx]->uv, 2 * sizeof(float));
+				t += texStride;
+			}
+			if (hasColor) {
+				memcpy(c, Vec4f::FromRGBA(points[idx]->color_32).AsArray(), 4 * sizeof(float));
```

Comment:
Dumb question: in theory, could we use `VK_FORMAT_R8G8B8A8_UNORM` for this, in some way, to communicate less data?  Or would that probably not help anyway (I guess especially with that padding issue)?

-[Unknown]

Reply:
Could stuff it in an uint32 and expand to float in the shader. Not worth the trouble, it's so little data anyway.

Reply:
Hmm, I thought Final Fantasy 4 used very high u/v, such that we had to keep increasing a buffer size, even though simple implementations just use 4x4 usually.  Maybe that was only generated points...?

Looks like the upper bound in size is about 3MB (65025 points - might be in PSP RAM more compressed using indices and s16.)

-[Unknown]

Reply:
IIRC that's it, it used crazy tesselation levels but not that much input data.

Haven't tried FF4 with this code, feel free to have a go :)

Reply:
Heh, there are a few obvious graphical issues in FF4, good thing I tried (not spline related.)

-[Unknown]

Reply:
Ah, indeed. It's a stencil issue, I got it.